### ISSUE TITLE
changes to handleQueryAction

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#53]
---

Author Name: **Chad Berkley** (Chad Berkley)
Original Redmine Issue: 57, https://projects.ecoinformatics.org/ecoinfo/issues/57
Original Date: 2000-08-09
Original Assignee: Chad Berkley

---

split handleQueryAction into three seperate functions.  One to create a query
(createQuery), another to create an squery (createSQuery) and another (runQuery)
to run the query and handle the results.  Any of these functions can be easily
overwritten for customized results.  handleQueryAction can call the appropriate
functions.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Chad Berkley** (Chad Berkley)
Original Date: 2000-08-14T21:40:59Z

---

I ended up breaking handleQueryAction into 4 different functions.
handleSQuery-handles structured queries
handleQuery-handles free text queries
runQuery-sends the created xml query to the DB for processing and returns the
results
transformDocument-takes the hashtable of results from runQuery and transforms
them to xml or html depending on the qformat param.
handlQueryAction is now just a stub from which these methods are called.

I also modified DBQuery to include a method for producing an xml structured
query from arbitrary CGI parameters.  This method is called createSQuery().
